tests/management/management/commands/test_utils.py

```diff
@@ -146,11 +148,21 @@ def test_import_service_account_data(self):
                     type=Principal.Types.SERVICE_ACCOUNT,
                     tenant=tenant_2,
                 ),
+                Principal(
+                    username=SERVICE_ACCOUNT_USERNAME_FORMAT.format(clientId=client_id_not_in_file),
+                    service_account_id=client_id_not_in_file,
+                    user_id=user_id_3,
+                    type=Principal.Types.SERVICE_ACCOUNT,
+                    tenant=tenant_2,
+                ),
             ]
         )
         with patch("builtins.open", mock_open(read_data=mock_file_content)):
             populate_service_account_data("file_name")
 
         for principal in principals:
             principal.refresh_from_db()
-            self.assertEqual(id_mapping[principal.service_account_id], principal.user_id)
+            if user_id := id_mapping.get(principal.service_account_id):
+                self.assertEqual(user_id, principal.user_id)
+            else:
+                self.assertEqual(user_id_3, principal.user_id)
```

Comment:
Nice fix! Quick thought – rather than using an id_mapping in the test like this, wdyt about testing each user_id, service_account_id pair directly? (Inspired by this blog post: https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html – the google testing blog has great stuff!)